libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -29,78 +32,114 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
   };
 }
 
+/*
+ * Provider Status
+ * defined here https://github.com/open-feature/js-sdk/blob/d26c4eeedde9cc321c69375594f1ccbfaff41a28/packages/shared/src/types.ts#L265
+ * ideally will be exported by '@openfeature/web-sdk'
+ * */
+enum ProviderStatus {
+  NOT_READY = 'NOT_READY',
+  READY = 'READY',
+  ERROR = 'ERROR',
+}
+
 export class LaunchDarklyClientProvider implements Provider {
   readonly metadata: ProviderMetadata = {
-    name: 'launchdarkly-client-provider',
+    name: 'launchdarkly-client',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  public status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  constructor(private readonly envKey: string, { providerLogger, ...ldOptions }: LaunchDarklyProviderOptions) {
```

Comment:
Though it's a breaking change, this is a much more idiomatic API. The previous one made sense before the events spec had fully matured, but now its ergonomics are not ideal

I suppose it would be possible to support the old one as well, but I question the value of doing so.